app/src/main/java/com/orange/ods/app/ui/components/Component.kt

```diff
@@ -54,6 +54,8 @@ sealed class Component(
 
     companion object {
         const val ImageBackgroundColor = 0xff1b1b1b
+
+        fun fromId(componentId: Long?) = components.firstOrNull { component -> component.id == componentId }
```

Comment:
`componentId` should not be nullable because a component with no ID does not really have any sense.

Same comment for variants.